File: src/ts/src/notification/NotificationBuilder.ts
export class NotificationBuilder {
    private _ticker: string;
    private _title: string;
    private _body: string;
    private _delay: number;
    private _interval: number;
    private _tag: number;

    public constructor() {
        this._ticker = ``;
        this._title = ``;
        this._body = ``;
        this._delay = 0;
        this._interval = 0;
        this._tag = 0;
    }

    public get ticker(): string {
        return this._ticker;
    }

    public set ticker(value: string) {
        this._ticker = value;
    }

    public get title(): string {
        return this._title;
    }

    public set title(value: string) {
        this._title = value;
    }

    public get body(): string {
        return this._body;
    }

    public set body(value: string) {
        this._body = value;
    }

    public get delay(): number {
        return this._delay;
    }

    public set delay(value: number) {
        this._delay = value;
    }

    public get interval(): number {
        return this._interval;
    }

    public set interval(value: number) {
        this._interval = value;
    }

    public get tag(): number {
        return this._tag;
    }

    public set tag(value: number) {
        this._tag = value;
    }
}

File: src/cpp/ee/notification/NotificationBuilder.cpp
//
//  NotificationBuilder.cpp
//  ee_x
//
//  Created by Zinge on 4/3/17.
//
//

#include "ee/notification/NotificationBuilder.hpp"

namespace ee {
namespace notification {
NotificationBuilder::NotificationBuilder()
    : ticker_("")
    , title_("")
    , body_("")
    , delay_(0)
    , interval_(0)
    , tag_(0) {}

NotificationBuilder& NotificationBuilder::setTicker(const std::string& ticker) {
    ticker_ = ticker;
    return *this;
}

NotificationBuilder& NotificationBuilder::setTitle(const std::string& title) {
    title_ = title;
    return *this;
}

NotificationBuilder& NotificationBuilder::setBody(const std::string& body) {
    body_ = body;
    return *this;
}

NotificationBuilder& NotificationBuilder::setDelay(int delay) {
    delay_ = delay;
    return *this;
}

NotificationBuilder& NotificationBuilder::setInterval(int interval) {
    interval_ = interval;
    return *this;
}

NotificationBuilder& NotificationBuilder::setTag(int tag) {
    tag_ = tag;
    return *this;
}

NotificationBuilder& NotificationBuilder::setStyle(int style) {
    style_ = style;
    return *this;
}
} // namespace notification
} // namespace ee
